server/py/services/api/crud/alert_activation.py

```diff
@@ -15,54 +15,117 @@
 
 from datetime import datetime, timedelta
 
+from sqlalchemy.orm import Session
+
 import mlrun.common.schemas.alert
 import mlrun.utils.singleton
+import services.api.db.sqldb.db
 
 
 class AlertActivation(
     metaclass=mlrun.utils.singleton.Singleton,
 ):
-    @staticmethod
-    def get_partition_info(
-        partition_interval: str,
-        partition_datetime: datetime,
-    ) -> tuple[str, str, str]:
+    def create_and_drop_partitions(
+        self,
+        session: Session,
+        retention_days: int,
+    ) -> None:
+        """
+        Creates partitions for the future based on the specified retention days
+        and drops old partitions that are older than the retention period.
+
+        :param session: SQLAlchemy session for database operations.
+        :param retention_days: The number of days to retain partitions.
         """
-        Generates partition details for a specified interval and datetime.
 
-        :param partition_interval: The partitioning interval type, e.g., "DAY", "MONTH", or "YEARWEEK".
-        :param partition_datetime: The datetime used for generating partition details.
+        # Retrieve the partition expression and interval from the database.
+        partition_expression, partition_interval = (
+            self.get_partition_expression_and_interval(session)
+        )
```

Comment:
Actually I now see that `partition_expression` is not used in this function at all.
But anyway you can get both values here and pass them to the `create` and `drop` functions